### ISSUE TITLE
Kraken wont train even with available xml files

### ISSUE THREAD

**Issue by Aarush786** [mittagessen/kraken#688]
Hi, 
I have been trying to train my hand writing model on kraken with ketos train command, I made a output.txt as a manifest file and then compiled it into dataset.arrow , even after running normally, it is only 1.3Kb and when used to train a model returns "No valid training data was provided to the train command. Please add
valid XML, line, or binary data."
Here's the full error: train.p │
│ y:393 in __init__                                                            │
│                                                                              │
│    390 │   │   │   self.train_set, self.val_set = random_split(train_set, (t │
│    391 │   │                                                                 │
│    392 │   │   if len(self.train_set) == 0 or len(self.val_set) == 0:        │
│ ❱  393 │   │   │   raise ValueError('No valid training data was provided to  │
│    394 │   │   │   │   │   │   │    'command. Please add valid XML, line, or │
│    395 │   │                                                                 │
│    396 │   │   if format_type == 'binary':                   

I am new in this and would appreciate any help. 

Thanks

**Comment by Aarush786** [comment]
<?xml version='1.0' encoding='utf-8'?>
<alto xmlns="http://www.loc.gov/standards/alto/ns-v4#"><Description><sourceImageInformation><fileName>../images/a01-091-06-06.png</fileName></sourceImageInformation></Description><Layout><Page ID="a01-091-06-06" PHYSICAL_IMG_NR="1" WIDTH="277" HEIGHT="114"><PrintSpace HPOS="1653" VPOS="1814" WIDTH="277" HEIGHT="114" TYPE="text"><ComposedBlockType ID="block_1" HPOS="1653" VPOS="1814" WIDTH="277" HEIGHT="114" TYPE="text"><TextBlock ID="textblock_1"><TextLine ID="line_1" HPOS="1653" VPOS="1814" WIDTH="277" HEIGHT="114"><String CONTENT="Brown" /></TextLine></TextBlock></ComposedBlockType></PrintSpace></Page></Layout></alto> 

This is one of my xml file

**Comment by mittagessen** [comment]
Your ALTO is missing all the baseline and bounding polygon information.

I think you're trying to create 'synthetic' ALTOs for a collection of
line strip images, right? This is a bad idea. If you want to train from
already cut out line images just use the legacy path format where the
images files share prefix up to the last extension with `.gt.txt` text
files containing your ground truth.

If you dont have a segmentation method independent of kraken, training a
model this way will be fairly useless though. Recognition models are
fairly sensitive to minor differences in segmentation output so a model
trained on one segmentation method's output won't perform well on
another's.

**Comment by Aarush786** [comment]
I am training Kraken with IMA datasets, and yes its of words which already cropped, and not lines. I am not sure of the use case of my model, as I am purely doing this to try different scenarios and learn from them. I havent got a lot of knowledge in this specific subject. But how I use the legacy method, you are referring to ?

**Comment by Aarush786** [comment]
I did try the way you are referring to, by creating images.gt truth files for tesseract but it didnt work, as tesseract is not that great in understanding htr

**Comment by mittagessen** [comment]
You're talking about IAM? There is a line-based version of it. Then just
write text files for each image line with a `.gt.txt` suffix and train.
Or if you're feeling fancy you can take their page-wise data and their
XML and transform it into ALTO. It is fairly straightforward.

**Comment by Aarush786** [comment]
Yes, my bad, IAM database. My first approach was to find the line database, however the website it is hosted on hasn't been working from a long time. Hence, I somehow found the word level segmentation on a old GitHub repo. I will try to check again if it's possible. The data I have has its own XML which refers to the file of each section in whole as a line. 
What command line do I use for training it directly with .gt.text of I am successful ?

**Comment by mittagessen** [comment]
Hm, it was available before Christmas but the group has been shut down a
couple of years ago so it might have finally broken.

On 25/02/19 05:00AM, Aarush786 wrote:
> What command line do I use for training it directly with .gt.text of I am successful ?

`ketos train -f path */*.png`

**Comment by Aarush786** [comment]
Thanks for your help, it did work at the end. However tested it with about 4000 files at the moment and the accuracy is 10 percent. Lets see if it improves on the way.
